src/vsr/replica.zig

```diff
@@ -1635,6 +1642,18 @@ pub fn ReplicaType(
             assert(message.header.replica < self.replica_count);
             assert(message.header.operation != .reserved);
 
+            // Sanity check --- if the prepare is definitely from the current log-wrap, it should be
+            // appended.
+            defer {
+                if (self.status == .normal and self.syncing == .idle and
+                    message.header.view == self.view and
+                    message.header.op > self.op_checkpoint() and
+                    message.header.op <= self.op_checkpoint_next_trigger())
```

Comment:
Could this be increased to
```suggestion
                    message.header.op <= self.op_prepare_max())
```
instead?

Reply:
I thought about that, but figured that by checking the trigger we are also indirectly checking that op_prepare_max itself is correct! It's a weaker invariant, but a separate code path.